Bug with SvelteKitCookies plugin - cannot find package '$app'

### Is this suited for github?

- [x] Yes, this is suited for github

### To Reproduce

Followed instructions as stated in 1.3 beta docs, `1.3.0-beta.1`, I get error 

`Cannot find package '$app' imported from /Users/xxxx/project/node_modules/.pnpm/better-auth@1.3.0-beta.1/node_modules/better-auth/dist/integrations/svelte-kit.mjs`


### Current vs. Expected behavior

Should work without issue.

### What version of Better Auth are you using?

1.3.0-beta.1

### Provide environment information

```bash
- MacOS 26
- Chrome v137.0.7151.120
```

### Which area(s) are affected? (Select all that apply)

Backend

### Auth config (if applicable)

```typescript

```

### Additional context

_No response_

Made it to stable yet this still appears to be the case? Also breaks @better-auth/cli in particular (`ERROR [Better Auth]: [#better-auth]: Couldn't read your auth config. error: Cannot find module '$app/server' ...`) when following the example in the docs.

> Made it to stable yet this still appears to be the case? Also breaks @better-auth/cli in particular (`ERROR [Better Auth]: [#better-auth]: Couldn't read your auth config. error: Cannot find module '$app/server' ...`) when following the example in the docs.

You have to pass `build` now and `getRequestEvent` now. The docs are accurate at this point - except I did `sveltekitCookies(async () => getRequestEvent())` instead of `sveltekitCookies(getRequestEvent)`

@jowsey I have the same issue with the CLI also

```
npx @better-auth/cli@latest generate  --config ./src/lib/server/auth/index.ts
2025-07-21T16:29:26.308Z ERROR [Better Auth]: Couldn't read your auth config. Error: Cannot find module '$lib/server/db'
```
Which is to be expected (?) since $lib is a typescript alias

Running it through `tsx` causes the same failure you are experiencing, I imagine since $app is a sveltekit alias (not sure of the exact mechanism).

```
pnpm tsx node_modules/@better-auth/cli/dist/index.mjs generate --config ./src/lib/server/auth/index.ts
2025-07-21T16:30:37.754Z ERROR [Better Auth]: Couldn't read your auth config. Error: Cannot find module '$app/server'
```

Perhaps this warrants a separate ticket?

--- a/docs/content/blogs/1-3.mdx
+++ b/docs/content/blogs/1-3.mdx
@@ -147,7 +147,7 @@ export const auth = betterAuth({
 Utilities for handling cookies in SvelteKit server actions.
 
 <Callout type="warn">
-  Breaking change: `building` and `getRequestEvent()` must now be passed in as props.
+  Breaking change: `building` and `getRequestEvent` must now be passed in as props.
 </Callout>
 
 ```ts title="auth.ts"

--- a/packages/better-auth/src/integrations/svelte-kit.ts
+++ b/packages/better-auth/src/integrations/svelte-kit.ts
@@ -77,16 +77,21 @@ export const sveltekitCookies = (
 							const event = getRequestEvent();
 							if (!event) return;
 							const parsed = parseSetCookieHeader(setCookies);
+
 							for (const [name, { value, ...ops }] of parsed) {
-								event.cookies.set(name, decodeURIComponent(value), {
-									sameSite: ops.samesite,
-									path: ops.path || "/",
-									expires: ops.expires,
-									secure: ops.secure,
-									httpOnly: ops.httponly,
-									domain: ops.domain,
-									maxAge: ops["max-age"],
-								});
+								try {
+									event.cookies.set(name, decodeURIComponent(value), {
+										sameSite: ops.samesite,
+										path: ops.path || "/",
+										expires: ops.expires,
+										secure: ops.secure,
+										httpOnly: ops.httponly,
+										domain: ops.domain,
+										maxAge: ops["max-age"],
+									});
+								} catch (e) {
+									// this will avoid any issue related to already streamed response
+								}
 							}
 						}
 					}),

--- a/packages/better-auth/src/integrations/svelte-kit.ts
+++ b/packages/better-auth/src/integrations/svelte-kit.ts
@@ -54,8 +54,9 @@ export function isAuthPath(url: string, options: BetterAuthOptions) {
 }
 
 export const sveltekitCookies = (
-	getRequestEvent: () => Promise<
-		RequestEvent<Partial<Record<string, string>>, string | null>
+	getRequestEvent: () => RequestEvent<
+		Partial<Record<string, string>>,
+		string | null
 	>,
 ) => {
 	return {
@@ -74,7 +75,7 @@ export const sveltekitCookies = (
 						if (returned instanceof Headers) {
 							const setCookies = returned?.get("set-cookie");
 							if (!setCookies) return;
-							const event = await getRequestEvent();
+							const event = getRequestEvent();
 							if (!event) return;
 							const parsed = parseSetCookieHeader(setCookies);
 							for (const [name, { value, ...ops }] of parsed) {

--- a/packages/better-auth/src/integrations/svelte-kit.ts
+++ b/packages/better-auth/src/integrations/svelte-kit.ts
@@ -28,7 +28,6 @@ export const svelteKitHandler = async ({
 	if (building) {
 		return resolve(event);
 	}
-
 	const { request, url } = event;
 	if (isAuthPath(url.toString(), auth.options)) {
 		return auth.handler(request);
@@ -78,6 +77,10 @@ export const sveltekitCookies = (
 							const event = getRequestEvent();
 							if (!event) return;
 							const parsed = parseSetCookieHeader(setCookies);
+							const routeId = event.route.id;
+							if (routeId) {
+								return;
+							}
 							for (const [name, { value, ...ops }] of parsed) {
 								event.cookies.set(name, decodeURIComponent(value), {
 									sameSite: ops.samesite,

--- a/packages/better-auth/src/integrations/svelte-kit.ts
+++ b/packages/better-auth/src/integrations/svelte-kit.ts
@@ -77,10 +77,6 @@ export const sveltekitCookies = (
 							const event = getRequestEvent();
 							if (!event) return;
 							const parsed = parseSetCookieHeader(setCookies);
-							const routeId = event.route.id;
-							if (routeId) {
-								return;
-							}
 							for (const [name, { value, ...ops }] of parsed) {
 								event.cookies.set(name, decodeURIComponent(value), {
 									sameSite: ops.samesite,

